### DISCUSSION TITLE
Cascade style All implementation

### DISCUSSION THREAD

**Discussion by forenheith** [nhibernate/nhibernate-core#3611] [Q&A]
Where can I find the source code that interprets the All invocation as the chain of calls listed here?
`/// <summary> save / delete / update / evict / lock / replicate / merge / persist</summary>
public static readonly CascadeStyle All = new AllCascadeStyle();`

**Comment by fredericDelaporte** [comment]
That is not a chain of call. It is a list of actions it will cascade. When one of the listed action occurs on the owning entity, that cascade style causes that action to be cascaded to the owned entity/entities. It does not trigger the other actions.

So, that is just:
https://github.com/nhibernate/nhibernate-core/blob/894529983457c110f2934904f27e2c3e9398c581/src/NHibernate/Engine/CascadeStyle.cs#L146-L152

Used there, by example:
https://github.com/nhibernate/nhibernate-core/blob/894529983457c110f2934904f27e2c3e9398c581/src/NHibernate/Engine/Cascade.cs#L97-L138
